docs/containers/testing-with-docker.md

```diff
@@ -0,0 +1,51 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Testing with docker](#testing-with-docker)
+  - [Using the included docker-compose support](#using-the-included-docker-compose-support)
+    - [Usage](#usage)
+    - [Configuration](#configuration)
+    - [Notes](#notes)
+    - [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Testing with docker
+
+## Using the included docker-compose support
+
+### Usage
+
+```bash
+docker build -t lyft/cartography
+docker-compose up -d
+docker-compose run cartography ...
+```
+
+### Configuration
+
+Configuration is possible via the `.compose` directory, which is
+git ignored. Environment for cartography is in
+`.compose/cartography/cartography.env`. neo4j config, logs, etc is
+located at `.compose/neo4j/...`
+
+### Notes
+
+* On initial start of the compose stack, it's necessary to
+change the neo4j user's password through the neo4j UI.
```

Comment:
[optional/I haven't tested in this context] Since this is strictly for dev testing, we could do something like this on startup to set the password to something:

```bash
if ((echo >/dev/tcp/localhost/7687) &>/dev/null); then echo "found neo4j" && curl -H "Content-Type: application/json" -X POST -d '{"password":"testtesttest"}' -u neo4j:neo4j http://localhost:7474/user/neo4j/password 
```

Reply:
I'm not opposed to this, but I'd like to followup with improvements like that in other PRs, based on other folks' feedback.